lib/thor.rb

```diff
@@ -238,6 +238,9 @@ def subcommand(subcommand, subcommand_class)
         args.unshift("help") if opts.include? "--help" or opts.include? "-h"
         invoke subcommand_class, args, opts, :invoked_via_subcommand => true, :class_options => options
       end
+      subcommand_class.commands.each do |meth, command|
```

Comment:
The workaround I found (and that, it seems, is generally used) is to explicitely set the subcommand's namespace, as

~~~
class MainTest < Thor
  namespace 'test'
end
~~~

I was wondering what are the upside/downside of your solution vs. doing

~~~
subcommand_class.namespace subcommand
~~~

in #subcommand